Bottom: 05a5cd3651f670e432b95b931b9a56acfd6820c7
Top:    971825c3c112e14db9db5ba78fbe42a1f9fde63d
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-26 16:57:50 -0700

Refresh of shut_wr

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 5fc5404..4eb1ce1 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -888,6 +888,8 @@ static int rs_poll_cq(struct rsocket *rs)
 				if (rs_msg_data(imm_data) == RS_CTRL_DISCONNECT) {
 					rs->state = rs_disconnected;
 					return ERR(ECONNRESET);
+				} else if (rs_msg_data(imm_data) == RS_CTRL_SHUTDOWN) {
+					rs->state &= ~rs_connect_rd;
 				}
 				break;
 			default:
